File: src/main/java/me/mrlion/lionplugin/commands/FlyCommand.java
package me.mrlion.lionplugin.commands;

import me.mrlion.lionplugin.LionPlugin;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Objects;

public class FlyCommand implements CommandExecutor {
    private final LionPlugin plugin;

    public FlyCommand(LionPlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!(sender instanceof Player player)) {
            sender.sendMessage(ChatColor.RED + "This command can't be used by console!");
            return true;
        }

        if (!player.hasPermission("lionplugin.fly")) {
            String noPermsMessage = Objects.requireNonNull(plugin.getConfig().getString("fly.no-permissions"));
            if (!noPermsMessage.isEmpty()) {
                player.sendMessage(ChatColor.translateAlternateColorCodes('&', noPermsMessage));
            } else {
                player.sendMessage(ChatColor.RED + "You don't have permission to use this command!");
            }
            return true;
        }

        if (player.getAllowFlight()) {
            player.setAllowFlight(false);
            plugin.getLogger().info(player.getName() + " has disabled flight mode.");
        } else {
            player.setAllowFlight(true);
            plugin.getLogger().info(player.getName() + " has enabled flight mode.");
        }

        return true;
    }
}

File: src/main/java/me/mrlion/lionplugin/commands/FlyCommand.kt
package me.mrlion.lionplugin.commands

import me.mrlion.lionplugin.LionPlugin
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*

class FlyCommand(private val plugin: LionPlugin) : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (sender !is Player) {
            sender.sendMessage(ChatColor.RED.toString() + "This command cant be used by console!")
            return true
        }

        if (!sender.hasPermission("lionplugin.fly")) {
            val noPermsMessage = Objects.requireNonNull(plugin.config.getString("fly.no-permissions"))?.let { ChatColor.translateAlternateColorCodes('&', it) }
            if (noPermsMessage != null) {
                if (noPermsMessage.isEmpty()) {
                    sender.sendMessage(ChatColor.RED.toString() + "You don't have permission to use this command!")
                } else {
                    sender.sendMessage(noPermsMessage)
                }
            }
            return true
        }

        if (sender.allowFlight) {
            sender.allowFlight = false
            plugin.logger.info(sender.name + " has disabled flight mode.")
        } else {
            sender.allowFlight = true
            plugin.logger.info(sender.name + " has enabled flight mode.")
        }

        return true
    }
}
